### PR TITLE
Realtime doc for lambda mode

### PR DESCRIPTION
**Pull request by hit-lacus**

### CODE REVIEW

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+}
+{% endhighlight %}
+
+In this sample, it is come from SSB with some additional field such as *event_time*. We have the field *event_time* as the timestamp of current event. 
+And we assumed that event came from countries of different timezone, "2019-12-09 08:44:50.000-0500" indicated that this a event which come from 'America/New_York' timezone. You may have some events which come from 'Asia/Shanghai' as well.
+
+*local_day_hour_minute* is a column which value is in local timezone, in this sample it in "GMT+8".
+
+### Question
+We want to do some realtime OLAP analysis, so you may consider to use Realtime OLAP. But you may have some concerns which included:
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: When perform realtime OLAP analysis with Kylin, you may have some concerns such as:

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+1. In the fact that events are come from different timezone, you may worried will this cause some trouble or incorrect query result?
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: Will events in different timezones cause incorrect query results?

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+2. In some cases, kafka message contains the value which is not actually what you want, says some dimension value is misspelled, how could you make corrections? (Or you want to retrieve some long-late-message which was dropped.)
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: 2. How could I make it correct when kafka messages contain the value which is not what you want, say ...
3. How could I retrieve long-late messages which has been dropped

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+3. My query only hit a small range of time range, how should I write filter condition make sure unused segments purged/skipped from scan?
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: time range -> time
make -> to make
are purged/skipped

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+### Quick Answer
+Firstly, you can always get the correct result in the right timezone of your place. Just by set *kylin.stream.event.timezone=GMT+N* for all Kylin processes. By default, UTC is used for *derived time column*.
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: you can always get the correct result in the right timezone of your location by set *kylin.stream.event.timezone=GMT+N* for all Kylin processes. By default, UTC is used for *derived time column*.

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+
+Secondly, in fact you cannot update a normal streaming cube, but you can update a streaming cube which in lambda mode, all you need to prepare is creating a Hive table which mapping to your kafka event.
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: mapping -> is mapped

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+
+Thirdly, yes it is, you can achieved this by add *derived time column* like *MINUTE_START*/*DAY_START* etc in your filter condition.
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: Thirdly, you can achieve it by adding *derived time column* in your filter condition like *MINUTE_START*, *DAY_START*, etc.

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+
+# How to do
+
+### Configure timezone
+We knew message may come from different timezone, but you want the query result should stick to some specific timezone.
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: knew -> know
but you want query results using some specific timezone.

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+
+Secondly, in fact you cannot update a normal streaming cube, but you can update a streaming cube which in lambda mode, all you need to prepare is creating a Hive table which mapping to your kafka event.
+
+Thirdly, yes it is, you can achieved this by add *derived time column* like *MINUTE_START*/*DAY_START* etc in your filter condition.
+
+# How to do
+
+### Configure timezone
+We knew message may come from different timezone, but you want the query result should stick to some specific timezone. 
+If you live in some place in GMT+2, please set *kylin.stream.event.timezone=GMT+2* for all Kylin process.
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: For example, if ...

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+# How to do
+
+### Configure timezone
+We knew message may come from different timezone, but you want the query result should stick to some specific timezone. 
+If you live in some place in GMT+2, please set *kylin.stream.event.timezone=GMT+2* for all Kylin process. 
+
+
+### Create lambda table
+
+You should create a hive table in *default* namespace, and this table should contains all column in your cube's dimension and measure, please
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: all your dimension and measure columns

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+
+### Configure timezone
+We knew message may come from different timezone, but you want the query result should stick to some specific timezone. 
+If you live in some place in GMT+2, please set *kylin.stream.event.timezone=GMT+2* for all Kylin process. 
+
+
+### Create lambda table
+
+You should create a hive table in *default* namespace, and this table should contains all column in your cube's dimension and measure, please 
+be ware of the *derived time column* like *MINUTE_START*/*DAY_START* which you are interested in should be also included if they are appeared in your cube's dimension column.
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: please remember to include *derived time column* like *MINUTE_START*/*DAY_START* if you set them in your cube's dimension column.

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+We knew message may come from different timezone, but you want the query result should stick to some specific timezone. 
+If you live in some place in GMT+2, please set *kylin.stream.event.timezone=GMT+2* for all Kylin process. 
+
+
+### Create lambda table
+
+You should create a hive table in *default* namespace, and this table should contains all column in your cube's dimension and measure, please 
+be ware of the *derived time column* like *MINUTE_START*/*DAY_START* which you are interested in should be also included if they are appeared in your cube's dimension column. 
+
+Depend on in which granularity level you want to update segment,  you can choose *HOUR_START* or *DAY_START* as partition column of this hive table.
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: Depending on which ..

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 13]
```diff
@@ -0,0 +1,174 @@
...
+,MONTH_START date
+)
+PARTITIONED BY (DAY_START date)
+STORED AS SEQUENCEFILE
+LOCATION 'hdfs:///LacusDir/lambda_flat_table';
+{% endhighlight %}
+
+
+### Create streaming cube in Kylin
+The first step is add information of broker list and topic name;
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: is to add kafka brokers and topic

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 14]
```diff
@@ -0,0 +1,174 @@
...
+PARTITIONED BY (DAY_START date)
+STORED AS SEQUENCEFILE
+LOCATION 'hdfs:///LacusDir/lambda_flat_table';
+{% endhighlight %}
+
+
+### Create streaming cube in Kylin
+The first step is add information of broker list and topic name; 
+after that, you should paste sample message into left and let Kylin auto-detect the column name and column type.
+You may find some data type is not corrected, please change to what its real data type and make sure it is aligned to Hive table.
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: not corrected -> correct
please fix them manually and make sure they are aligned to datatype in Hive table.

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 15]
```diff
@@ -0,0 +1,174 @@
...
+
+For example, you should change the data type of event_time from varchar to timestamp.
+And some column name is not the same as Hive Table, so please correct them too, such as from `customer_info_REGION` to `C_REGION`.
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: And some column names are...
so please correct them too, such as `customer_info_REGION` to `C_REGION`.

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 16]
```diff
@@ -0,0 +1,174 @@
...
+
+![image](/images/RealtimeOlap/Before-Submit.png)
+
+After that, please choose the right *TSColumn* *TSParser* and correct *Table Name*, table name should be equal to the name of Hive Table. After that, you should click *submit* buttom.
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: should be identical to the name of Hive Table

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 17]
```diff
@@ -0,0 +1,174 @@
...
+If you are luck enough, table meta info will be saved successfully, otherwise please correct data type and column name according to output message.
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: lucky

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 18]
```diff
@@ -0,0 +1,174 @@
...
+
+When you are creating Model, please set *Partition Date Column* in the right value. For streaming cube, *Partition Date Column* is used in generating HQL in updating segment which source data from Hive.
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: in the right value -> with the right value
*Partition Date Column* is used to generate HQL for updating segments whose source data is from Hive.

File: website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md [hunk 19]
```diff
@@ -0,0 +1,174 @@
...
+{% endhighlight %}
+
+![image](/images/RealtimeOlap/Timezone-checkresult.png)
+ 
+### Update segment
+
+1. Use some ETL tools like spark streaming to write correct data into HDFS, and add new partition based on your new data files. 
+2. After that, use Rest API *http://localhost:7070/kylin/api/cubes/{cube_name}/rebuild* [Put Method] to submit a build job to replace old segments, 
+please add offset according to timezone in *startTime* and *endTime* if you have set *kylin.stream.event.timezone*.
+3. In some case, you want add to a lot of historical data into Kylin (not replace something), you can also use the method.
```
Comment (Sean-Gu) [inline website/_docs30/tutorial/lambda_mode_and_timezone_realtime_olap.md]: you want to add..

**Comment by hit-lacus** [comment]
Thank you for your comments, I have updated my doc.

**Pull request merged**